tests/neg/implicitMatch-ambiguous.scala

```diff
@@ -0,0 +1,12 @@
+object Test {
+
+  class A
+  implicit val a1: A = new A
+  implicit val a2: A = new A
+
+  transparent def f: Any = implicit match {
+    case _: A => ???  // error: ambiguous implicits
```

Comment:
Could we report this error (and other inline errors) at call site instead? `f` may be defined in an external library.

Reply:
But if I write
```scala
class A

object Test  {
  def f(implicit a: A) = ???

  implicit val a1: A = new A
  implicit val a2: A = new A
  f // error
}
```
the error is reported at call site. Why would it be different?

> But then there is a line saying "this code was inlined at ..." where ... is the source and line of the callsite.

I believe we should do the opposite: report the issue at call site with "this line was inlined from ..." where ... is the source and line of the definition site. If you report the issue at definition site, your IDE might not be able to highlight anything.

In your example call site and definition site are in the same file, but most of the time it is not the case (and you might not have the source file of the definition, just a jar).

Reply:
> It is reported at the definition site so you can see what was ambiguous

Isn't it enough to report the ambiguous type in the error message? Highlighting the case does not seem to give you more information

Reply:
The problem is that many error messages refer to items of the code region that is highlighted. If we show the call site instead of the code region, these messages do not make sense. So in general we need to highlight the definition site and then can state from where it was inlined.